src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxeReferenceImpl.java

```diff
@@ -565,9 +565,11 @@ public boolean isReferenceTo(PsiElement element) {
       if (chain) return false;
     }
     final PsiElement resolve = element instanceof HaxeComponentName ? resolveToComponentName() : resolve();
+    boolean b = resolve instanceof HaxeNamedComponent;
+    boolean b1 = (resolve instanceof HaxeComponentName || b) &&
+                 resolve.getParent() instanceof HaxeClass;
     if (element instanceof HaxeFile &&
-        resolve instanceof HaxeComponentName &&
-        resolve.getParent() instanceof HaxeClass) {
+        (b1 || resolve instanceof AbstractHaxePsiClass) ) {
```

Comment:
b, b1 should be final at least.

I think `resolveToParents` and resolve instance of `HaxeComponentName && resolve.getParent() instanceof HaxeClass` didn't worked well together, because resolve to parent used to return something like

HaxeClass
HaxeFunctionDeclarationWithAttributes

For example, in testMoveFile1, IDE checks if some reference if it is reference to (`isReferenceTo`) some `element` which instance of `HaxeFile`, but it seems like to check it, we have to get `HaxeFile` of reference, this is what it used to be

```java
if (element instanceof HaxeFile && resolve instanceof HaxeComponentName && resolve.getParent() instanceof HaxeClass) { 
    return element == resolve.getContainingFile();
}
```

So in this case if we used `resolveToParents`, then instead of component name we may get `HaxeClass` which doesn't have a `HaxeClass` parent, so it fails.

In case if reference is resolved to `HaxeFunctionDeclarationWithAttributes`, it is probably not an instance of `HaxeComponentName`, so it fails.

I don't get this fully, but at least tests pass.